pkgs/build-support/rust/default.nix

```
@@ -60,7 +60,8 @@ let
 
   cargoDeps = if cargoVendorDir == null
     then fetchCargoTarball ({
-        inherit name src srcs sourceRoot unpackPhase cargoUpdateHook;
+        inherit src srcs sourceRoot unpackPhase cargoUpdateHook;
+        name = args.pname;
```

Not every package will have a pname, we can fall back to the `name` if it isn't set:

 ```suggestion
        name = args.pname or args.name;
```

Example:

```
nix-repl> args = { name = "foo"; }

nix-repl> args.pname or args.name  
"foo"
```